### PR TITLE
CIRC-319.1 Fix override scenarios. Renewal behavior for rolling loan policy with due date schedule

### PR DESCRIPTION
**Pull request by OleksiiMaksymov**
## Purpose
Changes in override scenarios
This resolves [CIRC-319](https://issues.folio.org/browse/CIRC-319)

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Code coverage on new code is 80% or greater
  - [x] Duplications on new code is 3% or less
  - [x] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.

### CODE REVIEW

File: src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java [hunk 1]
```diff
@@ -118,19 +118,11 @@ public static LoanPolicy unknown(String id) {
       final Result<DateTime> proposedDueDateResult =
         determineStrategy(true, systemDate).calculateDueDate(loan);
 
-      final JsonObject loansPolicy = getLoansPolicy();
-
-      if (proposedDueDateResult.failed() && isFixed(loansPolicy)) {
+      if (proposedDueDateResult.failed()) {
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java]: Does this mean that the decision to allow overrides is not dependent upon the type of policy?

File: src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java [hunk 2]
```diff
@@ -224,6 +208,10 @@ private boolean isSameOrBefore(Loan loan, DateTime proposedDueDate) {
       || proposedDueDate.isBefore(loan.getDueDate());
   }
 
+  private boolean numberOfRenewalsMatchLimit(Loan loan) {
```
Comment (marcjohnson-kint) [inline src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java]:
```suggestion
  private boolean hasReachedRenewalLimit(Loan loan) {
```

Comment (OleksiiMaksymov) [reply src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java]: Fix

Comment (OleksiiMaksymov) [reply src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java]: It seems to me that the expression `that limit has been reached` is more understandable. And this is not so misleading as the expressions: `has renewals remaining` and `has no renewals remaining`

File: src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java (hunk 1 above)
Comment (OleksiiMaksymov) [reply src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java]: In my understanding, on the basis of comments in CIRC-319 and correspondence with Emma, ​​yes.

Comment (marcjohnson-kint) [reply src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java]: Thanks

File: src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java (hunk 2 above)
Comment (marcjohnson-kint) [reply src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java]: How does `the limit has been reached` fit together with `there is no limit`?

I'm comfortable with that terminology if we can find a way to fit those together.

Comment (marcjohnson-kint) [reply src/main/java/org/folio/circulation/domain/policy/LoanPolicy.java]: I'm comfortable with the naming changes you have made

**Pull request merged**